src/modules/tools/zprobe/CalibrationStrategy.cpp

```diff
@@ -0,0 +1,597 @@
+// Generic calibration strategy based on nonlinear optimization
+//
+// Copyright (c) Oskar Linde 2015
+//
+// Licence: GPL v2
+
+#include "CalibrationStrategy.h"
+#include "Kernel.h"
+#include "Config.h"
+#include "Robot.h"
+#include "StreamOutputPool.h"
+#include "Gcode.h"
+#include "checksumm.h"
+#include "ConfigValue.h"
+#include "PublicDataRequest.h"
+#include "EndstopsPublicAccess.h"
+#include "PublicData.h"
+#include "Conveyor.h"
+#include "ZProbe.h"
+#include "BaseSolution.h"
+#include "StepperMotor.h"
+#include "Vector3.h"
+#include <cstddef>
+#include <fastmath.h>
+
+#define radius_checksum       CHECKSUM("radius")
+#define delta_calibration_strategy_checksum CHECKSUM("delta-calibration")
+
+std::function<void(float[3])> compute_rotation_transform(float p, float q, float offset, float inverse_r[3]);
```

Comment:
these need to be static so not to pollute the global namespace, in fact they should be private in the class.